test-app-legacy-context/app/src/MyComponent.js

```diff
@@ -82,6 +88,75 @@ export default ({ accounts }) => (
       </p>
       <strong>Single Device Data: </strong>
       <ContractData contract="ComplexStorage" method="singleDD" />
+      <strong>Array of UInts: </strong>
+      <ContractData contract="ComplexStorage" method="getuintarray" />{" "}
+    </div>
+    <div className="section">
+      <h2>ComplexStorage with Custom Rendering</h2>
+      <p>
+        This is the same data as above, but enhanced with a custom render
+        function.
+      </p>
+      <p>
+        <strong>String 1: </strong>
+        <ContractData
+          contract="ComplexStorage"
+          method="string1"
+          toUtf8
+          render={data => (
+            <>
+              This is the value: <b>{data}</b>
+            </>
+          )}
+        />
+      </p>
+      <p>
+        <strong>String 2: </strong>
+        <ContractData
+          contract="ComplexStorage"
+          method="string2"
+          toUtf8
+          render={myrender}
+        />
+      </p>
+      <strong>Single Device Data: </strong>
+      <ContractData
+        contract="ComplexStorage"
+        method="singleDD"
+        render={displayData => {
+          var i = 0;
+          const displayObjectProps = [];
+
+          Object.keys(displayData).forEach(key => {
+            if (i !== key) {
```

Comment:
Hmm I'm not really sure what's going on here. Was the intention to filter out the keys 0, 1, 2 (since they give the same values as keys deviceBrand, deviceYear, batteryWearLevel)?

Because I get the following output:

```
Single Device Data:
Element 0 has key: 0
and value: deviceBrand3
Element 1 has key: 1
and value: deviceYear3
Element 2 has key: 2
and value: wearLevel3
Element 3 has key: deviceBrand
and value: deviceBrand3
Element 4 has key: deviceYear
and value: deviceYear3
Element 5 has key: batteryWearLevel
and value: wearLevel3
```

Reply:
Eslint complained, changed it back.